src/sentry/tasks/deletion/hybrid_cloud.py

```diff
@@ -0,0 +1,214 @@
+    metrics.gauge(
+        "deletion.hybrid_cloud.low_bound",
+        value,
+        tags=dict(
+            table_name=field.model._meta.db_table,
+            field_name=field.name,
+            watermark=prefix,
+        ),
+    )
+
+
+def chunk_watermark_batch(
+    prefix: str, field: HybridCloudForeignKey, manager: Manager, *, batch_size: int
+) -> Tuple[int, int, bool, str]:
+    lower, transaction_id = get_watermark(prefix, field)
+    upper = manager.aggregate(Max("id"))["id__max"] or 0
+    batch_upper = min(upper, lower + batch_size)
+    return lower, upper, batch_upper + batch_size < upper, transaction_id
+
+
+@instrumented_task(
+    name="sentry.tasks.deletion.hybrid_cloud.schedule_hybrid_cloud_foreign_key_jobs",
+    queue="cleanup",
+    acks_late=True,
+)
+def schedule_hybrid_cloud_foreign_key_jobs():
+    for silo_mode in deletion_silo_modes():
+        for app, app_models in apps.all_models.items():
+            for model in app_models.values():
+                if not hasattr(model._meta, "silo_limit"):
+                    continue
+
+                # Only process models local this operational mode.
+                if silo_mode not in model._meta.silo_limit.modes:
+                    continue
+
+                for field in model._meta.fields:
+                    if not isinstance(field, HybridCloudForeignKey):
+                        continue
+
+                    process_hybrid_cloud_foreign_key_cascade_batch.delay(
+                        app_name=app,
+                        model_name=model.__name__,
+                        field_name=field.name,
+                        silo_mode=silo_mode.name,
+                    )
+
+
+@instrumented_task(
+    name="sentry.tasks.deletion.process_hybrid_cloud_foreign_key_cascade_batch",
+    queue="cleanup",
+    acks_late=True,
+)
+def process_hybrid_cloud_foreign_key_cascade_batch(
+    app_name: str, model_name: str, field_name: str, silo_mode: str
+) -> None:
+    try:
+        model = apps.get_model(app_label=app_name, model_name=model_name)
+        field: HybridCloudForeignKey
+        for field in model._meta.fields:
+            if not isinstance(field, HybridCloudForeignKey):
+                continue
+            if field.name == field_name:
+                break
+        else:
+            raise LookupError(f"Could not find field {field_name} on model {app_name}.{model_name}")
+
+        tombstone_cls: Any = TombstoneBase.class_for_silo_mode(SiloMode[silo_mode])
+
+        if _process_tombstone_reconcilition(
+            field, model, tombstone_cls, True
+        ) or _process_tombstone_reconcilition(field, model, tombstone_cls, False):
+            process_hybrid_cloud_foreign_key_cascade_batch.apply_async(
+                kwargs=dict(
+                    app_name=app_name,
+                    model_name=model_name,
+                    field_name=field_name,
+                    silo_mode=silo_mode,
+                ),
+                countdown=15,
```

Comment:
Why do you want to wait 15 seconds?

Reply:
Ideally there would be some more robust form of back pressure on the system so that spikes in deletions are better amortized.  15 seconds is "borrowed" from existing deletion logic, so I hoped it was a reasonable limit applied to the system.  This is definitely an area of future research (again, some form of manageable back pressure based on a constrained bandwidth is ideal).